### ISSUE TITLE
Any hopes for a windows compatible build?

### ISSUE THREAD

**Issue by devolution2409** [ButterflyStats/butterfly#5]
Hi  based Nox,
we're trying to make some app for windows that need to parse replays.. We tried compiling protobuff etc but to no avail..

I was wondering if had any plan to port this stuff to the unfriendly os that's windows?

thanks :)

**Comment by Noxville** [comment]
I had a chat with @spheenik about this today. In general, it is *probably* possible - but with quite a bit of work.

* mingw-w64 could possibly break on some templates
* might be some POSIX functions that act weirdly under MS runtime
* cross-compiling and linking external libraries will probably break a lot

Neither of us has the time to invest into getting that working really, but we'd be willing to help out where possible if someone really wanted to work on it. I think making another Docker build system is the easiest way to start.

**Comment by Philaeux** [comment]
We had troubles to even build dota protobuff+google protobuff on a windows system but I will focus onto it.

The plan is to build an offline solution to parse replays for coaching purposes (basically parse replays of scrims for a team that asked me to). The app is built over QT and C++ so it seems natural to use Butterfly for parsing, however the target has to Windows for obvious reasons...

**Comment by Noxville** [comment]
I'd suggest you use the built javascript bindings; since running a nodejs-like-process from C++ seems reasonable. Alternatively, make it web-based?

**Comment by Noxville** [comment]
(there are also Python-bindings, but I've actually never built them nor tested them)

**Comment by devolution2409** [comment]
That could be interesting indeed. 
I have no previous experience with emscripten whatsoever tho.
I'm guessing the necessary node modules are already compiled inside the dist folder?

**Comment by Noxville** [comment]
Yeah - I'll add a Dockerfile_js to the repo which can build the Javascript bindings; and here's an example of how it's used in the JS devkit.

**Comment by devolution2409** [comment]
I saw an example here: https://github.com/ButterflyStats/butterfly/blob/master/examples/js/example.js

`// you can get the packet IDs from the protobufs at <Link>`

'citation needed' as would wiki say :D

**Comment by Noxville** [comment]
https://github.com/SteamDatabase/GameTracking-Dota2/blob/master/Protobufs/dota_usermessages.proto#L98

**Comment by Noxville** [comment]
In https://github.com/ButterflyStats/butterfly/commit/1f14a282876824b19dd16cca7c29fb52a73ed9b6 I did a first stab at what the Javascript binding build would look like, just having some issues with snappy in the makefile. Perhaps @spheenik can help a bit - I won't have time for a day or two to look at this.
